### PR TITLE
Data stream lifecycle configures tail forcemerging

### PR DESCRIPTION
**Pull request by andreidan**
This adds support for tail merging in data stream lifecycle. Before issuing the forcemerge (without the target segment specified) the lifecycle service will aim to configure the floor segment merge policy to `100 MB` (configurable via a new cluster setting - `data_streams.lifecycle.target.merge.policy.floor_segment`) and the merge factor to `16` (configurable via a new cluster setting `data_streams.lifecycle.target.merge.policy.merge_factor`).

### CODE REVIEW

File: modules/dlm/src/test/java/org/elasticsearch/dlm/DataLifecycleServiceTests.java [hunk 1]
```diff
@@ -710,30 +730,6 @@ public void onFailure(Exception e) {
...
-        if (customDlmMetadata != null) {
-            indexMetadataBuilder.putCustom(LIFECYCLE_CUSTOM_INDEX_METADATA_KEY, customDlmMetadata);
-        }
-        indices.put(indexName, indexMetadataBuilder.build());
-        return ClusterState.builder(new ClusterName("test-cluster"))
-            .routingTable(routingTableBuilder.build())
-            .metadata(metadataBuilder.indices(indices).build())
-            .build();
-    }
-
```
Comment (andreidan) [inline modules/dlm/src/test/java/org/elasticsearch/dlm/DataLifecycleServiceTests.java:736]: This was moved below the public method

**Comment by andreidan** [comment]
Test failure was https://github.com/elastic/elasticsearch/issues/97018 which has been fixed in the meantime.

@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

**Review by gmarouli — APPROVED** [review]
LGTM! So curious to see this in action!

**Pull request merged**